src/pages/Landing/index.tsx

```diff
@@ -44,8 +44,28 @@ export default () => {
     // TODO: Stretch - Use the options object to handle errors.
   );
 
+  const newUserMutation = useMutation(postNewCustomer, {
+    onMutate: async newCustomer => {
+      await queryClient.cancelQueries('customers');
+      const previousCustomers = queryClient.getQueryData('customers');
+      queryClient.setQueryData('customers', old => [...old as Customer[], newCustomer]);
+      return { previousCustomers };
+    },
+    onError: (err, newCustomer, context) => {
+      queryClient.setQueryData('todos', context?.previousCustomers)
+    },
+    onSettled: () => queryClient.invalidateQueries({ queryKey: ['customers'] })
+  })
+
   const onSubmit = (e: FormEvent<Element>) => {
     e.preventDefault();
+    newUserMutation.mutate({
+      name: 'John Doe',
+      color: 'red',
+      age: 0,
+      isCool: false,
```

Comment:
Defining defaults should be done at the component level, if so desired. 
The issue with doing this here is you are spreading the "newUser" below these values.
Meaning any value that is NOT undefined (including "" or null) will overwrite these.

Reply:
Ooh, that's true, haven't thought of that. 😅